packages/core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,692 @@
+
+    if (!rangeMode) {
+      handleDateChange(event, newDate);
+      return;
+    }
+
+    if (position === "left") {
+      if (endDate) setStartDate(newDate > endDate ? endDate : newDate);
+    } else if (position === "right") {
+      if (!startDate) {
+        if (endDate) setStartDate(newDate > endDate ? endDate : newDate);
+        return;
+      }
+      setEndDate(newDate < startDate ? startDate : newDate);
+    }
+  };
+
+  /**
+   * Renders the container for the action elements.
+   */
+  const renderActions = () => (
+    <ClassNames>
+      {({ css }) => (
+        <HvActionBar
+          className={
+            showClear
+              ? clsx(
+                  datePickerClasses.actionContainer,
+                  classes?.actionContainer,
+                  css(styles.actionContainer)
+                )
+              : ""
+          }
+        >
+          {showClear && (
+            <div
+              className={clsx(
+                datePickerClasses.leftContainer,
+                classes?.leftContainer
+              )}
+            >
+              <HvButton
+                id={setId(id, "action", "clear")}
+                className={clsx(
+                  datePickerClasses.action,
+                  classes?.action,
+                  css(styles.action)
+                )}
+                variant="primaryGhost"
+                onClick={handleClear}
+              >
+                {labels?.clearLabel}
+              </HvButton>
+            </div>
+          )}
+          <div
+            className={clsx(
+              datePickerClasses.rightContainer,
+              classes?.rightContainer
+            )}
+          >
+            <HvButton
+              id={setId(id, "action", "apply")}
+              className={clsx(
+                datePickerClasses.action,
+                classes?.action,
+                css(styles.action)
+              )}
+              variant="primaryGhost"
+              onClick={handleApply}
+            >
+              {labels?.applyLabel}
+            </HvButton>
+            <HvButton
+              id={setId(id, "action", "cancel")}
+              className={clsx(
+                datePickerClasses.action,
+                classes?.action,
+                css(styles.action)
+              )}
+              variant="primaryGhost"
+              onClick={handleCancel}
+            >
+              {labels?.cancelLabel}
+            </HvButton>
+          </div>
+        </HvActionBar>
+      )}
+    </ClassNames>
+  );
+
+  const styledTypography = (dateString, variant, text) => {
+    const StyledTypography = styled(HvTypography)({
+      color: dateString
+        ? theme.colors.secondary
+        : theme.datePicker.dropdownPlaceholderColor,
+    });
+
+    return <StyledTypography variant={variant}>{text}</StyledTypography>;
+  };
+
+  const renderInput = (dateString) =>
+    styledTypography(
+      dateString,
+      activeTheme?.name == "ds3" ? "body" : "label",
```

Comment:
I think adding the `placeholderVariant` property to the themes like you had before was preferable in this case. This way, if someone uses a custom theme based on DS3 (and the theme name is not ds3), the `label` variant will always be used. However, instead of getting the CSS var `theme.datePicker.placeholderVariant` like before, we should do `activeTheme?.datePicker.placeholderVariant` to get the actual value. What do you think?

Reply:
Yes I agree, thank you , I'm going change it now